src/test/run-make-fulldeps/pgo-branch-weights/Makefile

```diff
@@ -15,21 +15,29 @@ ifdef IS_MSVC
 COMMON_FLAGS=-Cpanic=abort
 endif
 
+# For some very small programs GNU ld seems to not properly handle
+# instrumentation sections correctly. Neither Gold nor LLD have that problem.
+ifeq ($(UNAME),Linux)
+ifneq (,$(findstring x86,$(TARGET)))
+COMMON_FLAGS=-Clink-args=-fuse-ld=gold
```

Comment:
Hm, I think we're almost certain to have lld around (and can fairly quickly build it if needed, since we have caching in place and such) but I'm less sure about gold on CI. Maybe it makes sense to use lld here? If that doesn't work well locally and this passes CI though I'm fine with it.

Reply:
I thought that Gold is part of any standard binutils distribution?

Reply:
I guess we'll find out `:)`

Reply:
Hm, that's quite possible :) I had thought it would be more special (otherwise, why is ld the default?) but I guess there are edge cases and breaking the world is not an option for distros and such.

Reply:
> I thought that Gold is part of any standard binutils distribution?

This is true on most of Linux distributions but there are popular distros like openSUSE where gold is split to `binutils-gold` package and not installed by default AFAIK. This should be at least mentioned in the README so users know about this requirement.

Reply:
Yeah, it would be bad if this made run-make tests fail locally for some people. @Mark-Simulacrum, do we have a good way of using our own `rust-lld` in run-make tests?

> (otherwise, why is ld the default?)

Gold can only handle ELF, ld can handle at lot more, which is why, I'd guess.

Reply:
We add `lld` to the PATH of run-make tests if it's enabled (i.e., `lld_enabled` is set in config.toml). I imagine we could expose that similarly to `# needs-matching-clang` or so and ignore the test if lld is not built, and then make sure we're building lld on CI (we already do this on the `test-various` builer, I presume for WASM?). That should be a matter of adding `--set rust.lld` to `RUST_CONFIGURE_ARGS`, I think.

Reply:
Thanks for the info! I'll look into that in a follow-up PR if that's OK with you.

Reply:
That's fine with me, yes. I don't know that it's even necessary (maybe worth waiting until someone complains about the gold solution).